From a277b9216b09efe8ee4b04a66e43b235b4783354 Mon Sep 17 00:00:00 2001
From: Mark Mentovai <mark@mentovai.com>
Date: Thu, 28 Sep 2023 16:55:37 -0400
Subject: [PATCH] Remove std::unary_function usage

std::unary_function was removed with c++17

This is a backport of 82f7f3de0e90 to 5.15. The patch applies directly,
but the path to avfcamerautility.mm is different.
---
 src/plugins/avfoundation/camera/avfcamerautility.mm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git src/plugins/avfoundation/camera/avfcamerautility.mm src/plugins/avfoundation/camera/avfcamerautility.mm
index 25ccc4b01cc3..3f7f078bfc25 100644
--- src/plugins/avfoundation/camera/avfcamerautility.mm
+++ src/plugins/avfoundation/camera/avfcamerautility.mm
@@ -101,9 +101,9 @@ struct ByResolution
     }
 };
 
-struct FormatHasNoFPSRange : std::unary_function<AVCaptureDeviceFormat *, bool>
+struct FormatHasNoFPSRange
 {
-    bool operator() (AVCaptureDeviceFormat *format)
+    bool operator() (AVCaptureDeviceFormat *format) const
     {
         Q_ASSERT(format);
         return !format.videoSupportedFrameRateRanges || !format.videoSupportedFrameRateRanges.count;
-- 
2.42.0

